### PR TITLE
Ruby Feature [bug fix] : Fails if the dev container has the same ruby version installed

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Closes https://github.com/devcontainers/features/issues/601

### CODE REVIEW

File: test/ruby/scenarios.json [hunk 1]
```diff
@@ -13,5 +13,13 @@
         "features": {
             "ruby": {}
         }
+    },
+    "image_and_feature_installs_same_ruby_version": {
+        "image": "mcr.microsoft.com/devcontainers/universal:2.4.0",
```
Comment (joshspicer) [inline test/ruby/scenarios.json:18]: Is it necessary to use the universal image here?  If someone runs these tests locally (or if they're run in the pipeline) this large image will need to be pulled first)

File: src/ruby/install.sh [hunk 2]
```diff
@@ -177,9 +177,11 @@ fi
 find_version_from_git_tags RUBY_VERSION "https://github.com/ruby/ruby" "tags/v" "_"
 
 # Just install Ruby if RVM already installed
-if [ $(rvm --version) != "" ]; then
+if rvm --version > /dev/null; then
     echo "Ruby Version Manager already exists."
-    if [ "${RUBY_VERSION}" != "none" ]; then
+    if [[ "$(ruby -v)" = *"${RUBY_VERSION}"* ]]; then
```
Comment (joshspicer) [inline src/ruby/install.sh:182]: At this point I believe `RUBY_VERSION` is a fully expanded version (from line 177), right?

File: src/ruby/install.sh [hunk 3]
```diff
@@ -273,8 +278,10 @@ if [ "${SKIP_RBENV_RBUILD}" != "true" ]; then
             ln -s /usr/local/share/ruby-build /home/${USERNAME}/.rbenv/plugins/ruby-build
         fi
 
-        ln -s /usr/local/rvm/rubies/default/bin/ruby /usr/local/rvm/gems/default/bin 
-        
+        if [ ! -f /usr/local/rvm/gems/default/bin/ruby ]; then
+            ln -s /usr/local/rvm/rubies/default/bin/ruby /usr/local/rvm/gems/default/bin
```
Comment (joshspicer) [inline src/ruby/install.sh:282]: Do you know why we do this?

**Review by Princeprince559 — APPROVED** [review]

File: test/ruby/scenarios.json (hunk 1 above)
Comment (samruddhikhandale) [reply test/ruby/scenarios.json:18]: This issue only occurs with the universal image, as it uses this Ruby Feature to install a specific version of ruby. 😓 @joshspicer wondering, do you know of any other image which does that? 

I understand the pain point here, but this is the exact same scenario which is currently breaking.

File: src/ruby/install.sh (hunk 2 above)
Comment (samruddhikhandale) [reply src/ruby/install.sh:182]: Yes, that's correct.

image

File: src/ruby/install.sh (hunk 3 above)
Comment (samruddhikhandale) [reply src/ruby/install.sh:283]: Oryx expects ruby to be installed at  `/usr/local/rvm/gems/default/bin`, else it breaks the oryx flow of automatic installing gems. Hence, this bit was added. Added a comment!

File: test/ruby/scenarios.json (hunk 1 above)
Comment (samruddhikhandale) [reply test/ruby/scenarios.json]: The action is [running out of disk space](https://github.com/devcontainers/features/actions/runs/5627383080/job/15249820817#step:4:6294) (this is a manual run with trace logs which surfaced the cause) and failing due to the added test ^

Removing the test, validated locally that it works as expected 👇 : 

image

**Review by joshspicer — APPROVED** [review]

**Pull request merged**